src/game/editor/editor.cpp

```diff
@@ -3263,13 +3263,17 @@ void CEditor::RenderFileDialog()
 
 		if (m_FileDialogFileType == CEditor::FILETYPE_IMG && m_FilePreviewImage == 0 && m_FilesSelectedIndex > -1)
 		{
-			char aBuffer[1024];
-			str_format(aBuffer, sizeof(aBuffer), "%s/%s", m_pFileDialogPath, m_FileList[m_FilesSelectedIndex].m_aFilename);
-
-			if(Graphics()->LoadPNG(&m_FilePreviewImageInfo, aBuffer, IStorage::TYPE_ALL))
+			int Length = str_length(m_FileList[m_FilesSelectedIndex].m_aFilename);
+			if (Length >= 4 && !str_comp(m_FileList[m_FilesSelectedIndex].m_aFilename+Length-4, ".png"))
```

Comment:
You can use `str_endswith_nocase(m_FileList[m_FilesSelectedIndex].m_aFilename, ".png") != 0` instead